prov/rstream/src/rstream_ep.c

```diff
@@ -237,8 +237,8 @@ int rstream_ep_open(struct fid_domain *domain, struct fi_info *info,
 	rstream_ep->qp_win.ctrl_credits = RSTREAM_MAX_CTRL_TX;
 	rstream_ep->qp_win.max_rx_credits = rstream_info.rx_attr->size;
 
-	rstream_ep->tx_ctx.tx_ctxs = (struct fi_context *)
-		calloc(1, sizeof(struct fi_context) *
+	rstream_ep->tx_ctx.tx_ctxs = (struct rstream_ctx_data *)
+		calloc(1, sizeof(struct rstream_ctx_data) *
```

Comment:
No need to cast calloc() return.  In general, you should always try to use sizeof(some_field) in place of sizeof(struct blah).  This protects against code changes.  E.g. sizeof(*rstream_ep->tx_ctx.tx_ctxs).

Also,  calloc() takes as an argument the number of structures to allocate. Replace the '1' with rstream_ep->qp_win.max_tx_credits.